### PR TITLE
browser(webkit): roll to 23/09/21

### PR DESCRIPTION
**Pull request by dpino**
There were many conflicts in this update. Please take at look: https://github.com/dpino/WebKit/commit/486fca1914a47cfde12a53e7a975e59ee6fb5306

### CODE REVIEW

File: browser_patches/webkit/patches/bootstrap.diff [hunk 1]
```diff
@@ -5757,18 +5759,18 @@ index 16edb3bc689b8e2dde17597b642b706c1343e1f5..f363b2ca2410f22cff8d6ad908a88527
...
-+    rootGeometry.setPadding(Layout::Edges{});
-     rootGeometry.setBorder(Layout::Edges { { flow.borderStart(), flow.borderEnd() }, { flow.borderBefore(), flow.borderAfter() } });
-     rootGeometry.setHorizontalMargin({ });
-     rootGeometry.setVerticalMargin({ });
+@@ -258,7 +258,7 @@ void LineLayout::updateFormattingRootGeometryAndInvalidate()
+ 
+     auto updateGeometry = [&](auto& root) {
+         root.setContentBoxWidth(flow.contentLogicalWidth());
+-        root.setPadding(Layout::Edges { { flow.paddingStart(), flow.paddingEnd() }, { flow.paddingBefore(), flow.paddingAfter() } });
++	root.setPadding(Layout::Edges{});
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff:5770]: Wrong alignment?

File: browser_patches/webkit/patches/bootstrap.diff [hunk 2]
```diff
@@ -6434,7 +6436,7 @@ index 6a95f2aa052feeb07e38a7192d02d98468f7e5cb..b36ed11386c28cefc844901a0b57c45e
 +    if (element.isMouseFocusable())
 +        return true;
 +
-+    if (auto shadowRoot = makeRefPtr(element.shadowRoot())) {
++    if (RefPtr shadowRoot = element.shadowRoot()) {
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff:6439]: Does `RefPtr` without template parameter compile?

File: browser_patches/webkit/patches/bootstrap.diff [hunk 3]
```diff
@@ -9506,7 +9508,7 @@ index e4e88a11311aba49c65e6bc84767a1188588db36..33136b2448281b26c8c441626e38e536
 +{
 +    ASSERT(RunLoop::isMain());
 +
-+    m_queue->dispatch([this, protectedThis = makeRef(*this), sessionID, completionHandler = WTFMove(completionHandler)]() mutable {
++    m_queue->dispatch([this, protectedThis = Ref { *this }, sessionID, completionHandler = WTFMove(completionHandler)]() mutable {
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff:9511]: Why is this changing? Was `makeRef` removed from WebKit?

**Review by yury-s — APPROVED** [review]

**Comment by yury-s** [comment]
We have a couple of WebKit bugs that we want to fix in 1.15.x, let's not commit this roll for now as we want to minimize amount of browser changes in a patch release (we're going to build the new binary from master branch).

File: browser_patches/webkit/patches/bootstrap.diff (hunk 3 above)
Comment (dpino) [reply browser_patches/webkit/patches/bootstrap.diff:9511]: Right, `makeRef` and `makeRefPtr` were removed from WebKit codebase recently:
  - 2021-09-22 Drop makeRef() and use Ref { } instead https://bugs.webkit.org/show_bug.cgi?id=230626
  - 2021-09-22 Reduce use of makeRef() and use Ref { } instead https://bugs.webkit.org/show_bug.cgi?id=230585
  - 2021-09-21 Drop makeRefPtr() and use RefPtr { } directly instead https://bugs.webkit.org/show_bug.cgi?id=230564
  - 2021-09-21 Drop remaining uses of makeRefPtr() in WebCore/ https://bugs.webkit.org/show_bug.cgi?id=230527
  - 2021-09-20 Reduce use of makeRefPtr() and use RefPtr { } directly https://bugs.webkit.org/show_bug.cgi?id=230503
  - 2021-09-20 Stop using makeRef(*this) / makeRefPtr(this) https://bugs.webkit.org/show_bug.cgi?id=230464

File: browser_patches/webkit/patches/bootstrap.diff (hunk 2 above)
Comment (dpino) [reply browser_patches/webkit/patches/bootstrap.diff:6439]: Yes, it builds. I built with the changes applied on Ubuntu 20.04 and Ubuntu 18.04. So it should build on both systems.

Regarding the change, I took a look at the changes applied in the set of patches that removed makeRef/makeRefPtr. Here's a very similar change: https://github.com/WebKit/webkit/compare/102a6d5be26fa%5E..102a6d5be26fa#diff-37167271e633e42f6b0cb277c8eb74680c5515cb6eef47bb4321897e5f5314e9L400 from bug Drop remaining uses of makeRefPtr() in WebCore/ https://bugs.webkit.org/show_bug.cgi?id=230527

**Comment by yury-s** [comment]
We resolved the webkit issues affecting current release, let's land this roll.

**Pull request merged**